src/v/datalake/tests/arrow_writer_test.cc

```diff
@@ -0,0 +1,107 @@
+    schema.fields.emplace_back(
+      nested_field::create(3, "test_long", required, long_type{}));
+    schema.fields.emplace_back(
+      nested_field::create(4, "test_float", required, float_type{}));
+    schema.fields.emplace_back(
+      nested_field::create(5, "test_double", required, double_type{}));
+    schema.fields.emplace_back(
+      nested_field::create(6, "test_decimal", required, decimal_type{8, 8}));
+    schema.fields.emplace_back(
+      nested_field::create(7, "test_date", required, date_type{}));
+    schema.fields.emplace_back(
+      nested_field::create(8, "test_time", required, time_type{}));
+    schema.fields.emplace_back(
+      nested_field::create(9, "test_timestamp", required, timestamp_type{}));
+    schema.fields.emplace_back(nested_field::create(
+      10, "test_timestamptz", required, timestamptz_type{}));
+    schema.fields.emplace_back(
+      nested_field::create(11, "test_string", required, string_type{}));
+    schema.fields.emplace_back(
+      nested_field::create(12, "test_uuid", required, uuid_type{}));
+    schema.fields.emplace_back(
+      nested_field::create(13, "test_fixed", required, fixed_type{100}));
+    schema.fields.emplace_back(
+      nested_field::create(14, "test_binary", required, binary_type{}));
+
+    struct_type nested_schema;
+    nested_schema.fields.emplace_back(
+      nested_field::create(1, "nested_bool", required, boolean_type{}));
+    nested_schema.fields.emplace_back(
+      nested_field::create(2, "nested_int", required, int_type{}));
+    nested_schema.fields.emplace_back(
+      nested_field::create(3, "nested_long", required, long_type{}));
+
+    schema.fields.emplace_back(nested_field::create(
+      15, "test_struct", required, std::move(nested_schema)));
+
+    schema.fields.emplace_back(nested_field::create(
+      16,
+      "test_list",
+      required,
+      list_type::create(1, required, string_type{})));
+
+    schema.fields.push_back(nested_field::create(
+      17,
+      "test_map",
+      required,
+      map_type::create(1, string_type{}, 2, required, long_type{})));
+    return schema;
+}
+
+TEST(ArrowWriterTest, TranslatesSchemas) {
+    datalake::arrow_writer writer(test_schema(iceberg::field_required::no));
+    auto schema = writer.build_arrow_schema();
+    ASSERT_NE(schema, nullptr);
+    std::string expected_schema
+      = "test_bool: bool\ntest_int: int32\ntest_long: int64\ntest_float: "
+        "float\ntest_double: double\ntest_decimal: decimal128(8, "
+        "8)\ntest_date: date32[day]\ntest_time: time64[us]\ntest_timestamp: "
+        "timestamp[us]\ntest_timestamptz: timestamp[us, tz=UTC]\ntest_string: "
+        "string\ntest_uuid: fixed_size_binary[16]\ntest_fixed: "
+        "fixed_size_binary[100]\ntest_binary: binary\ntest_struct: "
+        "struct<nested_bool: bool, nested_int: int32, nested_long: "
+        "int64>\ntest_list: list<element: string>\ntest_map: map<string, "
+        "int64>";
```

Comment:
nit: might be easier to read with raw values eg

```
auto explected_schema = R"(test_bool: bool
test_int: int32
test_log: int64
...
)";
```

Reply:
Done